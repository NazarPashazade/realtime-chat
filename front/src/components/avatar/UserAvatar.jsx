import { Avatar } from '@mui/material';
import React from 'react';
import { deepOrange, deepPurple } from '@mui/material/colors';
import styles from './UserAvatar.module.scss'

export default function UserAvatar({ user }) {
    const { avatarURL, username } = user

    return (
        <>
            {avatarURL
                ? <Avatar className={styles.avatar} alt={username} src={avatarURL} />
                : <Avatar className={styles.avatar} sx={{ bgcolor: deepOrange[500] }}>{username[0]}</Avatar>
            }
        </>
    );
}

