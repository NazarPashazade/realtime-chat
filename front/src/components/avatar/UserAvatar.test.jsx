import renderer from 'react-test-renderer'
import UserAvatar from './UserAvatar'

describe('Renders avatar component', () => {
    it('snapshot test', () => {
        const mockUser = { avatarURL: "", username: "" }
        const tree = renderer.create(<UserAvatar user={mockUser} />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})
