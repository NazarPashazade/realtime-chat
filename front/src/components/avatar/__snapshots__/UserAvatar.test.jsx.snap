// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders avatar component snapshot test 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-1d3jhju-MuiAvatar-root"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
    data-testid="PersonIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
  </svg>
</div>
`;
